openmc/mgxs_library.py

```diff
@@ -0,0 +1,705 @@
+            subelement = ET.SubElement(element, 'representation')
+            subelement.text = self._representation
+
+        if self._representation == 'angle':
+            if self._num_azimuthal is not None:
+                subelement = ET.SubElement(element, 'num_azimuthal')
+                subelement.text = str(self._num_azimuthal)
+            if self._num_polar is not None:
+                subelement = ET.SubElement(element, 'num_polar')
+                subelement.text = str(self._num_polar)
+
+        if self._scatt_type is not None:
+            subelement = ET.SubElement(element, 'scatt_type')
+            subelement.text = self._scatt_type
+
+        if self._order is not None:
+            subelement = ET.SubElement(element, 'order')
+            subelement.text = str(self._order)
+
+        if self._tabular_legendre is not None:
+            subelement = ET.SubElement(element, 'tabular_legendre')
+            subelement.set('enable', str(self._tabular_legendre['enable']))
+            subelement.set('num_points', str(self._tabular_legendre['num_points']))
+
+        if self._total is not None:
+            subelement = ET.SubElement(element, 'total')
+            subelement.text = ndarray_to_string(self._total)
+
+        if self._absorption is not None:
+            subelement = ET.SubElement(element, 'absorption')
+            subelement.text = ndarray_to_string(self._absorption)
+
+        if self._scatter is not None:
+            subelement = ET.SubElement(element, 'scatter')
+            subelement.text = ndarray_to_string(self._scatter)
+
+        if self._multiplicity is not None:
+            subelement = ET.SubElement(element, 'multiplicity')
+            subelement.text = ndarray_to_string(self._multiplicity)
+
+        if self._fissionable:
+            if self._fission is not None:
+                subelement = ET.SubElement(element, 'fission')
+                subelement.text = ndarray_to_string(self._fission)
+
+            if self._k_fission is not None:
+                subelement = ET.SubElement(element, 'k_fission')
+                subelement.text = ndarray_to_string(self._k_fission)
+
+            if self._nu_fission is not None:
+                subelement = ET.SubElement(element, 'nu_fission')
+                subelement.text = ndarray_to_string(self._nu_fission)
+
+            if self._chi is not None:
+                subelement = ET.SubElement(element, 'chi')
+                subelement.text = ndarray_to_string(self._chi)
+
+        return element
+
+class MGXSLibraryFile(object):
+    """Multi-Group Cross Sections file used for an OpenMC simulation.
+    Corresponds directly to the MG version of the cross_sections.xml input file.
+
+    Attributes
+    ----------
+    energy_groups : openmc.mgxs.EnergyGroups
+        Energy group structure.
+    inverse_velocities : Iterable of Real
+        Inverse of velocities, units of sec/cm
+    filename : str
+        XML file to write to.
+    """
```

Comment:
I got the xsdata object, but the cross sections file is 'private' so i dont think i should be declaring that in the docstring, right?